### ISSUE TITLE
Add ability to set default world gen

### ISSUE THREAD

**Issue by foxanthony1** [PG85/OpenTerrainGenerator#114]
Basically to give modpack creators the ability take the choice away from players regarding world gen (especially the seed) like Default World Generator does, which unfortunately doesn't seem to work with OTG.

It appears that OTG supersedes DWG maybe that could be corrected? With OTG installed DWG's options simply don't appear but OTG's do and it should be the opposite.

I guess the biggest issue is players spawning in deep water so often, especially since I'm removing their ability to swim. I'm sure that's something you all are working on but I expect it's not going to be fixed soon.

**Comment by foxanthony1** [comment]
Oh, wait... I get it. You don't really want this in modpacks yet. That's why the config files are in the /mods folder.

I'm slow, but I usually get there eventually. I did figure out how to set a default seed though.

**Comment by PG85** [comment]
Hey, sorry, not up to date with all the abbreviations for mods, what's DWG?

You can package your configs as a worldpack btw, just like Biome Bundle and the OTG example worlds. A world pack is essentially a seperate mod jar that contains only OTG configs and a mod description, it auto-installs the OTG configs when MC starts. You'll need to grab the worldpacker project on git and edit a few lines to make a jar file though, that requires a bit of technical knowledge (see: https://github.com/PG85/WorldPacker).

The deep water bug is known, will take a look at it in the future, it's not high on the prio list atm though. It's a bit unfortunate but it can be easily avoided by picking a good seed.

**Comment by foxanthony1** [comment]
Sorry, I shortened Default World Generator ( [](https://www.curseforge.com/minecraft/mc-mods/default-world-generator-port) ) to DWG, it's a tool for modpack devs to control the world generation. Players are unable to change the seed or anything else in the world generation.

If I make a mod with my own settings and seed, many players will still mess with things and end up with bad spawns. They'll complain and need their hands held to get a proper world.

Streamers and YouTubers regularly use their own seeds to generate worlds. How do I prevent that? They won't even play my pack after they drown in the first 10 worlds they generate (my world only has 3 biome types, I spent a whole day trying to find a good seed).

OTG's world gen has problems with the water but it's also spectacular if you get the right seed. Give modpack devs the opportunity to show off OTG at it's best by allowing them to lock down a seed that the players can't casually change and will show just how amazing the mod is.
